Src/Couchbase.Linq/UnixMillisecondsDateTime.cs

```diff
@@ -0,0 +1,117 @@
+        {
+            if (obj.GetType() != typeof (UnixMillisecondsDateTime))
+            {
+                return false;
+            }
+
+            return _dateTime == ((UnixMillisecondsDateTime) obj)._dateTime;
+        }
+
+        public override int GetHashCode()
+        {
+            return _dateTime.GetHashCode();
+        }
+
+        public override string ToString()
+        {
+            // ReSharper disable once SpecifyACultureInStringConversionExplicitly
+            return _dateTime.ToString();
+        }
+
+        public static UnixMillisecondsDateTime FromDateTime(DateTime dateTime)
+        {
+            return new UnixMillisecondsDateTime(dateTime);
+        }
+
+        public static UnixMillisecondsDateTime? FromDateTime(DateTime? dateTime)
+        {
+            if (dateTime.HasValue)
+            {
+                return new UnixMillisecondsDateTime(dateTime.Value);
+            }
+            else
+            {
+                return null;
+            }
+        }
+
+        public static DateTime ToDateTime(UnixMillisecondsDateTime unixMillisecondsDateTime)
+        {
+            return unixMillisecondsDateTime._dateTime;
+        }
+
+        public static DateTime? ToDateTime(UnixMillisecondsDateTime? unixMillisecondsDateTime)
+        {
+            if (unixMillisecondsDateTime.HasValue)
+            {
+                return unixMillisecondsDateTime.Value._dateTime;
+            }
+            else
+            {
+                return null;
+            }
+        }
+
+        public static bool operator ==(UnixMillisecondsDateTime left, UnixMillisecondsDateTime right)
+        {
+            return left._dateTime == right._dateTime;
+        }
+
+        public static bool operator !=(UnixMillisecondsDateTime left, UnixMillisecondsDateTime right)
+        {
+            return left._dateTime != right._dateTime;
+        }
+
+        public static bool operator >(UnixMillisecondsDateTime left, UnixMillisecondsDateTime right)
+        {
+            return left._dateTime > right._dateTime;
+        }
+
+        public static bool operator <(UnixMillisecondsDateTime left, UnixMillisecondsDateTime right)
+        {
+            return left._dateTime < right._dateTime;
+        }
+
+        public static bool operator >=(UnixMillisecondsDateTime left, UnixMillisecondsDateTime right)
+        {
+            return left._dateTime >= right._dateTime;
+        }
+
+        public static bool operator <=(UnixMillisecondsDateTime left, UnixMillisecondsDateTime right)
+        {
+            return left._dateTime <= right._dateTime;
+        }
+
+        //public static implicit operator UnixMillisecondsDateTime(DateTime dateTime)
+        //{
+        //    return FromDateTime(dateTime);
+        //}
+
+        //public static implicit operator DateTime(UnixMillisecondsDateTime unixMillisecondsDateTime)
+        //{
+        //    return ToDateTime(unixMillisecondsDateTime);
+        //}
```

Comment:
Should remove this or move to a separate branch (if you want to keep it for a later commit).